File: orderservice/src/main/java/com/fon/njt/orderservice/service/PaymentService.java
package com.fon.njt.orderservice.service;

import com.fon.njt.orderservice.dto.PaymentIntentDto;
import com.stripe.exception.StripeException;
import com.stripe.model.PaymentIntent;

public interface PaymentService {
    PaymentIntent paymentIntent(PaymentIntentDto paymentIntentDto) throws StripeException;

    PaymentIntent confirm(String id) throws StripeException;

    PaymentIntent cancel(String id) throws StripeException;
}

File: client-app/src/app/cart/payment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { Observable } from 'rxjs';
import { PaymentIntent } from './payment-intent.model';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {

  constructor(private http:HttpClient) { }

  public pay(paymentIntentDto: PaymentIntent):Observable<string>{
    return this.http.post<string>(`${environment.apiUrl}/book-orders/stripe/payment-intent`, paymentIntentDto)
  }

  public confirm(id: string):Observable<any>{
    return this.http.post<any>(`${environment.apiUrl}/book-orders/stripe/confirm/${id}`, {});
  }

  public cancel(id: string):Observable<string>{
    return this.http.post<string>(`${environment.apiUrl}/book-orders/stripe/cancel/${id}`, {});
  }
}
